packages/python/src/components/PyDoc.tsx

```diff
@@ -4,239 +4,230 @@ import { ParameterDescriptor } from "../parameter-descriptor.js";
 import { resolveTypeExpression } from "../utils.js";
 import { Atom, type TypeExpressionProps } from "./index.js";
 
-interface GoogleStyleDocParamTypeProps {
-  type?: TypeExpressionProps;
-  default?: boolean;
-}
-
-function GoogleStyleDocParamType(props: GoogleStyleDocParamTypeProps) {
-  const resolvedType =
-    props.type ? resolveTypeExpression(props.type) : undefined;
-  return (
-    <>
-      <Show when={Boolean(props.type)}>
-        {" ("}
-        {resolvedType}
-        <Show when={props.default}>{", optional"}</Show>
-        {")"}
-      </Show>
-    </>
-  );
-}
-
-interface GoogleStyleDocParamNameProps {
-  name: Children;
-}
-
-function GoogleStyleDocParamName(props: GoogleStyleDocParamNameProps) {
-  return <>{props.name}</>;
-}
-
-interface GoogleStyleDocParamDescriptionProps {
-  children?: Children;
-  default?: Children;
-}
-
-function GoogleStyleDocParamDescription(
-  props: GoogleStyleDocParamDescriptionProps,
-) {
-  return (
-    <Show when={Boolean(props.children)}>
-      {": "}
-      <align width={4}>
-        <Prose>
-          {props.children}
-          <Show when={Boolean(props.default)}>
-            <br />
-            Defaults to <Atom jsValue={props.default} />.
-          </Show>
-        </Prose>
-      </align>
-    </Show>
-  );
-}
-
-export interface GoogleStyleDocParamProps {
-  name: Children;
-  type?: TypeExpressionProps;
-  children?: Children;
-  default?: Children;
+export interface FunctionDocProps {
+  description: Children[];
+  parameters?: ParameterDescriptor[];
+  returns?: string;
+  yields?: string;
+  raises?: string[];
+  note?: string;
+  style?: "google";
 }
 
 /**
- * Create a GoogleStyleDoc parameter.
+ * A component that creates a FunctionDoc block for functions.
  */
-export function GoogleStyleDocParam(props: GoogleStyleDocParamProps) {
-  return (
-    <>
-      <GoogleStyleDocParamName name={props.name} />
-      <GoogleStyleDocParamType
-        type={props.type}
-        default={Boolean(props.default)}
-      />
-      <GoogleStyleDocParamDescription
-        children={props.children}
-        default={props.default}
+export function FunctionDoc(props: FunctionDocProps) {
+  const style = props.style ?? "google";
+  if (style === "google") {
```

Comment:
nit: I would make these `switch` statements, as the future intent is that we do something different for another value of `style`.